sentinelhub/sentinelhub_request.py

```diff
@@ -87,52 +84,61 @@ def create_request(self):
         )]
 
     @staticmethod
-    def input_data(data_source=None, time_interval=None, maxcc=1.0, mosaicking_order='mostRecent', other_args=None):
+    def input_data(data_collection=None, time_interval=None, maxcc=None, mosaicking_order=None, other_args=None,
+                   data_source=None):
         """ Generate the `input` part of the Processing API request body
 
-        :param data_source: One of supported ProcessingAPI data sources.
-        :type data_source: sentinelhub.DataSource
+        :param data_collection: One of supported ProcessingAPI data sources.
+        :type data_collection: DataCollection
         :param time_interval: interval with start and end date of the form YYYY-MM-DDThh:mm:ss or YYYY-MM-DD
         :type time_interval: (str, str) or (datetime, datetime)
         :param maxcc: Maximum accepted cloud coverage of an image. Float between 0.0 and 1.0. Default is 1.0.
-        :type maxcc: float
+        :type maxcc: float or None
         :param mosaicking_order: Mosaicking order, which has to be either 'mostRecent', 'leastRecent' or 'leastCC'.
-        :type mosaicking_order: str
+        :type mosaicking_order: str or None
         :param other_args: Additional dictionary of arguments. If provided, the resulting dictionary will get updated
                            by it.
         :param other_args: dict
+        :param data_source: A deprecated alternative of data_collection
+        :type data_source: DataCollection
         """
+        data_collection = data_source or data_collection
+        if data_source is not None:
+            warnings.warn('Parameter data_source is deprecated, use data_collection instead',
+                          category=SHDeprecationWarning)
+        if not isinstance(data_collection, DataCollection):
```

Comment:
Why is not such check done also in other similar cases?

Reply:
I now added casting `DataCollection(data_collection)` everywhere where needed. This has more or less the same effect than the check above.